packages/kotti-table/src/logic/column.js

```diff
@@ -17,25 +17,40 @@ export const defaultState = {
 export const mutations = {
 	insertColumn(store, { column, index }) {
 		const deleted = store.state._destroyedColumns[column.prop]
-		setColumn(store.state, { column, index, deleted })
+		setColumn(store.state, {
+			column,
+			index,
+			deleted,
+		})
 		deleted && (store.state._destroyedColumns[column.prop] = 0)
-		store.commit('updateColumns', { emitChange: false })
+		store.commit('updateColumns', {
+			emitChange: false,
+		})
 	},
 	removeColumn(store, column) {
 		column = getColumn(store.state, column)
 		if (column) {
 			column._deleted = true
 			store.state._destroyedColumns[column.prop] = 1
-			store.commit('updateColumns', { emitChange: false })
+			store.commit('updateColumns', {
+				emitChange: false,
+			})
 		}
 	},
 	setColumns(store, columns = store.state._columnsArray) {
 		const { state } = store
 		const pickedColumns = columns.map(col => pick(col, PUBLIC_COLUMN_PROPS))
-		for (const col of pickedColumns) {
+		for (let i = 0; i < pickedColumns.length; i++) {
+			const col = pickedColumns[i]
 			const column = getColumn(state, col)
-			if (column) {
+
+			if (column && state._columns.length > 0) {
 				Object.assign(column, col)
+			} else {
```

Comment:
(2) It looks unintentional that the else clause fires on column resolving to falsy

Reply:
it is intentional. If the column does not exist in the old state, it will set a new column from pickedColumns

Reply:
Discussed in person: the extra restriction is to prevent corrupted state to modify the table. This is correct.